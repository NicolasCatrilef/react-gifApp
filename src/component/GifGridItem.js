import React from 'react'

export const GifGridItem = ( { id, title, url } ) => {
    
    // console.log( id, title, url );
    
    return (
        <div className = "card animate__animated animate__bounce">
            {/* { img.titule } */}
            <img src = { url } alt = { title } />
           <p> { title } </p> 
        </div>
    )
}
