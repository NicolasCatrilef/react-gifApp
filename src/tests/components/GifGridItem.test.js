import { shallow } from "enzyme";
import { GifGridItem } from "../../component/GifGridItem";

describe('Pruebas en Gif Grid Item', () => {

    const title = 'Prueba Apps';
    const url = 'http://localhost/prueba.png';
    const wrapper = shallow( <GifGridItem title = { title } url = { url } /> );
    
    test('Renderizar el GifGridItem', () => {
        expect( wrapper ).toMatchSnapshot();
    });
    

    test('Parrafo debe tener un valor', () => {
        const p = wrapper.find( 'p' );
        
        expect( p.text().trim() ).toBe( title );
    })
    

    test('Pruebas sobre imagenes', () => {
        const img = wrapper.find( 'img' );
        
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
    })

    test('Debe tener la clase CSS', () => {
        const div = wrapper.find( 'div' );
        const cn = div.prop('className');
        
        expect( cn.includes( 'animate__bounce' ) ).toBe( true );
    })
    
    
});
 