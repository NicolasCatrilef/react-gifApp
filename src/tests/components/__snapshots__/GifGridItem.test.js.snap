// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Gif Grid Item Renderizar el GifGridItem 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Prueba Apps"
    src="http://localhost/prueba.png"
  />
  <p>
     
    Prueba Apps
     
  </p>
</div>
`;
